### ISSUE TITLE
Segmentation faults with large arrays

### ISSUE THREAD

**Issue by Spivoxity** [Spivoxity/obc-3#8]
Diego Sardina writes: This code with a large array (of 2^18-1 elements) leads to segmentation fault.
```
MODULE test2;
  VAR string: ARRAY 0FFFFFFFH OF CHAR; i: INTEGER;
BEGIN
  FOR i := 0 TO LEN(string)-1 DO END;
END test2.
```
Platform: Linux CentOS 7.3 x64

**Comment by Spivoxity** [comment]
The problem is artithmetic overflow of pointer arithmetic in gc.c(scratch_alloc).

**Comment by Spivoxity** [comment]
As a workaround, try allocating from the heap:
```
VAR string: POINTER TO ARRAY OF CHAR;
NEW(string, 0FFFFFFFFH);
```

**Comment by Spivoxity** [comment]
Fixed in 3.0.3
